ipns.go

```diff
@@ -130,7 +134,7 @@ func createCborDataForIpnsEntry(e *pb.IpnsEntry) ([]byte, error) {
 func Validate(pk ic.PubKey, entry *pb.IpnsEntry) error {
 	// Check the ipns record signature with the public key
 
-	// Check v2 signature if it's available, otherwise use the v1 signature
+	// Check v2 signature if it's available
 	if entry.GetSignatureV2() != nil {
```

Comment:
You can remove most of the nested code by doing:

``` go
if entry.GetSignatureV2() == nil {
	// always error if no valid signature could be found
	return ErrSignature
}
```

I think is easier to read.